
<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Ag3 API &#8212; malariagen_data API v7.0.0 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=ac02cc09edc035673794" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=ac02cc09edc035673794" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=ac02cc09edc035673794" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=ac02cc09edc035673794" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=a746c00c" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=ac02cc09edc035673794" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=ac02cc09edc035673794" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=ac02cc09edc035673794"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js?v=62f2956d"></script>
    <script src="_static/doctools.js?v=888ff710"></script>
    <script src="_static/sphinx_highlight.js?v=4825356b"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'Ag3';</script>
    <script>
        DOCUMENTATION_OPTIONS.theme_version = '0.14.1';
        DOCUMENTATION_OPTIONS.theme_switcher_json_url = 'https://leehart.github.io/tmp_test_6/switcher.json';
        DOCUMENTATION_OPTIONS.theme_switcher_version_match = 'v7.0.0';
        DOCUMENTATION_OPTIONS.show_version_warning_banner = false;
        </script>
    <link rel="icon" href="_static/logo.svg"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="malariagen_data.ag3.Ag3.pca" href="generated/malariagen_data.ag3.Ag3.pca.html" />
    <link rel="prev" title="malariagen_data API" href="index.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search the docs ..."
         aria-label="Search the docs ..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
<div class="bd-header__inner bd-page-width">
  <label class="sidebar-toggle primary-toggle" for="__primary">
    <span class="fa-solid fa-bars"></span>
  </label>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.svg" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="_static/logo.svg" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<script>
document.write(`
  <div class="version-switcher__container dropdown">
    <button id="versionswitcherbutton" type="button" role="button" class="version-switcher__button btn btn-sm navbar-btn dropdown-toggle" data-bs-toggle="dropdown" aria-haspopup="listbox" aria-controls="versionswitcherlist" aria-label="Version switcher list">
      Choose version  <!-- this text may get changed later by javascript -->
      <span class="caret"></span>
    </button>
    <div id="versionswitcherlist" class="version-switcher__menu dropdown-menu list-group-flush py-0" role="listbox" aria-labelledby="versionswitcherbutton">
    <!-- dropdown will be populated by javascript on page load -->
    </div>
  </div>
`);
</script></div>
      
        <div class="navbar-item"><nav class="navbar-nav">
  <p class="sidebar-header-items__title"
     role="heading"
     aria-level="1"
     aria-label="Site Navigation">
    Site Navigation
  </p>
  <ul class="bd-navbar-elements navbar-nav">
    
                    <li class="nav-item current active">
                      <a class="nav-link nav-internal" href="#">
                        Ag3 API
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="Af1.html">
                        Af1 API
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="Amin1.html">
                        Amin1 API
                      </a>
                    </li>
                
  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          
 <script>
 document.write(`
   <button class="btn navbar-btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
        </div>
      
      
        <div class="navbar-item">
<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script></div>
      
        <div class="navbar-item"><ul class="navbar-icon-links navbar-nav"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/malariagen/malariagen-data-python" title="GitHub" class="nav-link" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><span><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i></span>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">
 <script>
 document.write(`
   <button class="btn navbar-btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script>
    </div>
  

  
    <label class="sidebar-toggle secondary-toggle" for="__secondary" tabindex="0">
      <span class="fa-solid fa-outdent"></span>
    </label>
  
</div>

    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          <div class="navbar-item">
<script>
document.write(`
  <div class="version-switcher__container dropdown">
    <button id="versionswitcherbutton" type="button" role="button" class="version-switcher__button btn btn-sm navbar-btn dropdown-toggle" data-bs-toggle="dropdown" aria-haspopup="listbox" aria-controls="versionswitcherlist" aria-label="Version switcher list">
      Choose version  <!-- this text may get changed later by javascript -->
      <span class="caret"></span>
    </button>
    <div id="versionswitcherlist" class="version-switcher__menu dropdown-menu list-group-flush py-0" role="listbox" aria-labelledby="versionswitcherbutton">
    <!-- dropdown will be populated by javascript on page load -->
    </div>
  </div>
`);
</script></div>
        
          <div class="navbar-item"><nav class="navbar-nav">
  <p class="sidebar-header-items__title"
     role="heading"
     aria-level="1"
     aria-label="Site Navigation">
    Site Navigation
  </p>
  <ul class="bd-navbar-elements navbar-nav">
    
                    <li class="nav-item current active">
                      <a class="nav-link nav-internal" href="#">
                        Ag3 API
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="Af1.html">
                        Af1 API
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="Amin1.html">
                        Amin1 API
                      </a>
                    </li>
                
  </ul>
</nav></div>
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">
<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script></div>
        
          <div class="navbar-item"><ul class="navbar-icon-links navbar-nav"
    aria-label="Icon Links">
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/malariagen/malariagen-data-python" title="GitHub" class="nav-link" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><span><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i></span>
            <span class="sr-only">GitHub</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item"><nav class="bd-docs-nav bd-links"
     aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">Section Navigation</p>
  <div class="bd-toc-item navbar-nav"><ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.pca.html">malariagen_data.ag3.Ag3.pca</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_pca_coords.html">malariagen_data.ag3.Ag3.plot_pca_coords</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_pca_coords_3d.html">malariagen_data.ag3.Ag3.plot_pca_coords_3d</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_pca_variance.html">malariagen_data.ag3.Ag3.plot_pca_variance</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.releases.html">malariagen_data.ag3.Ag3.releases</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.results_cache_get.html">malariagen_data.ag3.Ag3.results_cache_get</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.results_cache_set.html">malariagen_data.ag3.Ag3.results_cache_set</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.v3_wild.html">malariagen_data.ag3.Ag3.v3_wild</a></li>
</ul>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aim_calls.html">malariagen_data.ag3.Ag3.aim_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aim_variants.html">malariagen_data.ag3.Ag3.aim_variants</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cohort_diversity_stats.html">malariagen_data.ag3.Ag3.cohort_diversity_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.count_samples.html">malariagen_data.ag3.Ag3.count_samples</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cross_metadata.html">malariagen_data.ag3.Ag3.cross_metadata</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_aim_heatmap.html">malariagen_data.ag3.Ag3.plot_aim_heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_samples_interactive_map.html">malariagen_data.ag3.Ag3.plot_samples_interactive_map</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.sample_cohorts.html">malariagen_data.ag3.Ag3.sample_cohorts</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.species_calls.html">malariagen_data.ag3.Ag3.species_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.wgs_data_catalog.html">malariagen_data.ag3.Ag3.wgs_data_catalog</a></li>
</ul>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aa_allele_frequencies.html">malariagen_data.ag3.Ag3.aa_allele_frequencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aa_allele_frequencies_advanced.html">malariagen_data.ag3.Ag3.aa_allele_frequencies_advanced</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.diversity_stats.html">malariagen_data.ag3.Ag3.diversity_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.geneset.html">malariagen_data.ag3.Ag3.geneset</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.genome_features.html">malariagen_data.ag3.Ag3.genome_features</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.genome_sequence.html">malariagen_data.ag3.Ag3.genome_sequence</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.igv.html">malariagen_data.ag3.Ag3.igv</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.is_accessible.html">malariagen_data.ag3.Ag3.is_accessible</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_genome.html">malariagen_data.ag3.Ag3.open_genome</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_site_annotations.html">malariagen_data.ag3.Ag3.open_site_annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_snp_genotypes.html">malariagen_data.ag3.Ag3.open_snp_genotypes</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_diversity_stats.html">malariagen_data.ag3.Ag3.plot_diversity_stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_heterozygosity.html">malariagen_data.ag3.Ag3.plot_heterozygosity</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_heterozygosity_track.html">malariagen_data.ag3.Ag3.plot_heterozygosity_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_heatmap.html">malariagen_data.ag3.Ag3.plot_frequencies_heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_interactive_map.html">malariagen_data.ag3.Ag3.plot_frequencies_interactive_map</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_map_markers.html">malariagen_data.ag3.Ag3.plot_frequencies_map_markers</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_time_series.html">malariagen_data.ag3.Ag3.plot_frequencies_time_series</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_genes.html">malariagen_data.ag3.Ag3.plot_genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_snps.html">malariagen_data.ag3.Ag3.plot_snps</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_snps_track.html">malariagen_data.ag3.Ag3.plot_snps_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_transcript.html">malariagen_data.ag3.Ag3.plot_transcript</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.resolve_region.html">malariagen_data.ag3.Ag3.resolve_region</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.site_annotations.html">malariagen_data.ag3.Ag3.site_annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.site_filters.html">malariagen_data.ag3.Ag3.site_filters</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_allele_counts.html">malariagen_data.ag3.Ag3.snp_allele_counts</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_allele_frequencies.html">malariagen_data.ag3.Ag3.snp_allele_frequencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_allele_frequencies_advanced.html">malariagen_data.ag3.Ag3.snp_allele_frequencies_advanced</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_calls.html">malariagen_data.ag3.Ag3.snp_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_dataset.html">malariagen_data.ag3.Ag3.snp_dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_effects.html">malariagen_data.ag3.Ag3.snp_effects</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_genotypes.html">malariagen_data.ag3.Ag3.snp_genotypes</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_variants.html">malariagen_data.ag3.Ag3.snp_variants</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_sites.html">malariagen_data.ag3.Ag3.snp_sites</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.view_alignments.html">malariagen_data.ag3.Ag3.view_alignments</a></li>
</ul>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.h12_calibration.html">malariagen_data.ag3.Ag3.h12_calibration</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.h12_gwss.html">malariagen_data.ag3.Ag3.h12_gwss</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.h1x_gwss.html">malariagen_data.ag3.Ag3.h1x_gwss</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.haplotypes.html">malariagen_data.ag3.Ag3.haplotypes</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_haplotypes.html">malariagen_data.ag3.Ag3.open_haplotypes</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_haplotype_sites.html">malariagen_data.ag3.Ag3.open_haplotype_sites</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h12_calibration.html">malariagen_data.ag3.Ag3.plot_h12_calibration</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h12_gwss_track.html">malariagen_data.ag3.Ag3.plot_h12_gwss_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h12_gwss.html">malariagen_data.ag3.Ag3.plot_h12_gwss</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h1x_gwss.html">malariagen_data.ag3.Ag3.plot_h1x_gwss</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h1x_gwss_track.html">malariagen_data.ag3.Ag3.plot_h1x_gwss_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_haplotype_clustering.html">malariagen_data.ag3.Ag3.plot_haplotype_clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_haplotype_network.html">malariagen_data.ag3.Ag3.plot_haplotype_network</a></li>
</ul>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cnv_coverage_calls.html">malariagen_data.ag3.Ag3.cnv_coverage_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cnv_discordant_read_calls.html">malariagen_data.ag3.Ag3.cnv_discordant_read_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cnv_hmm.html">malariagen_data.ag3.Ag3.cnv_hmm</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.gene_cnv.html">malariagen_data.ag3.Ag3.gene_cnv</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.gene_cnv_frequencies.html">malariagen_data.ag3.Ag3.gene_cnv_frequencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.gene_cnv_frequencies_advanced.html">malariagen_data.ag3.Ag3.gene_cnv_frequencies_advanced</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_cnv_coverage_calls.html">malariagen_data.ag3.Ag3.open_cnv_coverage_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_cnv_discordant_read_calls.html">malariagen_data.ag3.Ag3.open_cnv_discordant_read_calls</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_cnv_hmm.html">malariagen_data.ag3.Ag3.open_cnv_hmm</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage.html">malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage_track.html">malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap.html">malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap_track.html">malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_roh.html">malariagen_data.ag3.Ag3.plot_roh</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_roh_track.html">malariagen_data.ag3.Ag3.plot_roh_track</a></li>
<li class="toctree-l1"><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.roh_hmm.html">malariagen_data.ag3.Ag3.roh_hmm</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item">



<nav aria-label="Breadcrumb">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a href="index.html" class="nav-link" aria-label="Home">
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
    <li class="breadcrumb-item active" aria-current="page">Ag3 API</li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section id="ag3-api">
<h1>Ag3 API<a class="headerlink" href="#ag3-api" title="Permalink to this heading">#</a></h1>
<p>This page provides a curated list of functions and properties available in the malariagen_data API relating to <em>Anopheles gambiae</em> data.</p>
<section id="basic-data-access">
<h2>Basic data access<a class="headerlink" href="#basic-data-access" title="Permalink to this heading">#</a></h2>
<table class="autosummary longtable table autosummary">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.pca.html#malariagen_data.ag3.Ag3.pca" title="malariagen_data.ag3.Ag3.pca"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pca</span></code></a>(region, n_snps[, thin_offset, ...])</p></td>
<td><p>Run a principal components analysis (PCA) using biallelic SNPs from the selected genome region and samples.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_pca_coords.html#malariagen_data.ag3.Ag3.plot_pca_coords" title="malariagen_data.ag3.Ag3.plot_pca_coords"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_pca_coords</span></code></a>(data[, x, y, color, symbol, ...])</p></td>
<td><p>Plot sample coordinates from a principal components analysis (PCA) as a plotly scatter plot.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_pca_coords_3d.html#malariagen_data.ag3.Ag3.plot_pca_coords_3d" title="malariagen_data.ag3.Ag3.plot_pca_coords_3d"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_pca_coords_3d</span></code></a>(data[, x, y, z, color, ...])</p></td>
<td><p>Plot sample coordinates from a principal components analysis (PCA) as a plotly 3D scatter plot.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_pca_variance.html#malariagen_data.ag3.Ag3.plot_pca_variance" title="malariagen_data.ag3.Ag3.plot_pca_variance"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_pca_variance</span></code></a>(evr[, width, height])</p></td>
<td><p>Plot explained variance ratios from a principal components analysis (PCA) using a plotly bar plot.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.releases.html#malariagen_data.ag3.Ag3.releases" title="malariagen_data.ag3.Ag3.releases"><code class="xref py py-obj docutils literal notranslate"><span class="pre">releases</span></code></a></p></td>
<td><p>The releases for which data are available at the given storage location.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.results_cache_get.html#malariagen_data.ag3.Ag3.results_cache_get" title="malariagen_data.ag3.Ag3.results_cache_get"><code class="xref py py-obj docutils literal notranslate"><span class="pre">results_cache_get</span></code></a>(*, name, params)</p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.results_cache_set.html#malariagen_data.ag3.Ag3.results_cache_set" title="malariagen_data.ag3.Ag3.results_cache_set"><code class="xref py py-obj docutils literal notranslate"><span class="pre">results_cache_set</span></code></a>(*, name, params, results)</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.v3_wild.html#malariagen_data.ag3.Ag3.v3_wild" title="malariagen_data.ag3.Ag3.v3_wild"><code class="xref py py-obj docutils literal notranslate"><span class="pre">v3_wild</span></code></a></p></td>
<td><p>Legacy, convenience property to access sample sets from the 3.0 release, excluding the lab crosses.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="sample-metadata-access">
<h2>Sample metadata access<a class="headerlink" href="#sample-metadata-access" title="Permalink to this heading">#</a></h2>
<table class="autosummary longtable table autosummary">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aim_calls.html#malariagen_data.ag3.Ag3.aim_calls" title="malariagen_data.ag3.Ag3.aim_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">aim_calls</span></code></a>(aims[, sample_sets, sample_query])</p></td>
<td><p>Access ancestry informative marker SNP sites, alleles and genotype calls.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aim_variants.html#malariagen_data.ag3.Ag3.aim_variants" title="malariagen_data.ag3.Ag3.aim_variants"><code class="xref py py-obj docutils literal notranslate"><span class="pre">aim_variants</span></code></a>(aims)</p></td>
<td><p>Open ancestry informative marker variants.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cohort_diversity_stats.html#malariagen_data.ag3.Ag3.cohort_diversity_stats" title="malariagen_data.ag3.Ag3.cohort_diversity_stats"><code class="xref py py-obj docutils literal notranslate"><span class="pre">cohort_diversity_stats</span></code></a>(cohort, cohort_size, ...)</p></td>
<td><p>Compute genetic diversity summary statistics for a cohort of individuals.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.count_samples.html#malariagen_data.ag3.Ag3.count_samples" title="malariagen_data.ag3.Ag3.count_samples"><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_samples</span></code></a>([sample_sets, sample_query, ...])</p></td>
<td><p>Create a pivot table showing numbers of samples available by space, time and taxon.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cross_metadata.html#malariagen_data.ag3.Ag3.cross_metadata" title="malariagen_data.ag3.Ag3.cross_metadata"><code class="xref py py-obj docutils literal notranslate"><span class="pre">cross_metadata</span></code></a>()</p></td>
<td><p>Load a dataframe containing metadata about samples in colony crosses, including which samples are parents or progeny in which crosses.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_aim_heatmap.html#malariagen_data.ag3.Ag3.plot_aim_heatmap" title="malariagen_data.ag3.Ag3.plot_aim_heatmap"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_aim_heatmap</span></code></a>(aims[, sample_sets, ...])</p></td>
<td><p>Plot a heatmap of ancestry-informative marker (AIM) genotypes.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_samples_interactive_map.html#malariagen_data.ag3.Ag3.plot_samples_interactive_map" title="malariagen_data.ag3.Ag3.plot_samples_interactive_map"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_samples_interactive_map</span></code></a>([sample_sets, ...])</p></td>
<td><p>Plot an interactive map showing sampling locations using ipyleaflet.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.sample_cohorts.html#malariagen_data.ag3.Ag3.sample_cohorts" title="malariagen_data.ag3.Ag3.sample_cohorts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">sample_cohorts</span></code></a>([sample_sets])</p></td>
<td><p>Access cohorts metadata for one or more sample sets.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.species_calls.html#malariagen_data.ag3.Ag3.species_calls" title="malariagen_data.ag3.Ag3.species_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">species_calls</span></code></a>([sample_sets])</p></td>
<td><p>Access species calls for one or more sample sets.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.wgs_data_catalog.html#malariagen_data.ag3.Ag3.wgs_data_catalog" title="malariagen_data.ag3.Ag3.wgs_data_catalog"><code class="xref py py-obj docutils literal notranslate"><span class="pre">wgs_data_catalog</span></code></a>(sample_set)</p></td>
<td><p>Load a data catalog providing URLs for downloading BAM, VCF and Zarr files for samples in a given sample set.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="snp-data-access">
<h2>SNP data access<a class="headerlink" href="#snp-data-access" title="Permalink to this heading">#</a></h2>
<table class="autosummary longtable table autosummary">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aa_allele_frequencies.html#malariagen_data.ag3.Ag3.aa_allele_frequencies" title="malariagen_data.ag3.Ag3.aa_allele_frequencies"><code class="xref py py-obj docutils literal notranslate"><span class="pre">aa_allele_frequencies</span></code></a>(transcript, cohorts[, ...])</p></td>
<td><p>Compute per amino acid allele frequencies for a gene transcript.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.aa_allele_frequencies_advanced.html#malariagen_data.ag3.Ag3.aa_allele_frequencies_advanced" title="malariagen_data.ag3.Ag3.aa_allele_frequencies_advanced"><code class="xref py py-obj docutils literal notranslate"><span class="pre">aa_allele_frequencies_advanced</span></code></a>(transcript, ...)</p></td>
<td><p>Group samples by taxon, area (space) and period (time), then compute amino acid change allele counts and frequencies.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.diversity_stats.html#malariagen_data.ag3.Ag3.diversity_stats" title="malariagen_data.ag3.Ag3.diversity_stats"><code class="xref py py-obj docutils literal notranslate"><span class="pre">diversity_stats</span></code></a>(cohorts, cohort_size, ...[, ...])</p></td>
<td><p>Compute genetic diversity summary statistics for multiple cohorts.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.geneset.html#malariagen_data.ag3.Ag3.geneset" title="malariagen_data.ag3.Ag3.geneset"><code class="xref py py-obj docutils literal notranslate"><span class="pre">geneset</span></code></a>(*args, **kwargs)</p></td>
<td><p>Deprecated, this method has been renamed to genome_features().</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.genome_features.html#malariagen_data.ag3.Ag3.genome_features" title="malariagen_data.ag3.Ag3.genome_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">genome_features</span></code></a>([region, attributes])</p></td>
<td><p>Access genome feature annotations (e.g.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.genome_sequence.html#malariagen_data.ag3.Ag3.genome_sequence" title="malariagen_data.ag3.Ag3.genome_sequence"><code class="xref py py-obj docutils literal notranslate"><span class="pre">genome_sequence</span></code></a>(region[, inline_array, chunks])</p></td>
<td><p>Access the reference genome sequence.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.igv.html#malariagen_data.ag3.Ag3.igv" title="malariagen_data.ag3.Ag3.igv"><code class="xref py py-obj docutils literal notranslate"><span class="pre">igv</span></code></a>(region[, tracks])</p></td>
<td><p>Create an IGV browser and display it within the notebook.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.is_accessible.html#malariagen_data.ag3.Ag3.is_accessible" title="malariagen_data.ag3.Ag3.is_accessible"><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_accessible</span></code></a>(region, site_mask)</p></td>
<td><p>Compute genome accessibility array.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_genome.html#malariagen_data.ag3.Ag3.open_genome" title="malariagen_data.ag3.Ag3.open_genome"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_genome</span></code></a>()</p></td>
<td><p>Open the reference genome zarr.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_site_annotations.html#malariagen_data.ag3.Ag3.open_site_annotations" title="malariagen_data.ag3.Ag3.open_site_annotations"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_site_annotations</span></code></a>()</p></td>
<td><p>Open site annotations zarr.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_snp_genotypes.html#malariagen_data.ag3.Ag3.open_snp_genotypes" title="malariagen_data.ag3.Ag3.open_snp_genotypes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_snp_genotypes</span></code></a>(sample_set)</p></td>
<td><p>Open SNP genotypes zarr.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_diversity_stats.html#malariagen_data.ag3.Ag3.plot_diversity_stats" title="malariagen_data.ag3.Ag3.plot_diversity_stats"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_diversity_stats</span></code></a>(df_stats[, color, ...])</p></td>
<td><p>Plot diversity statistics.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_heterozygosity.html#malariagen_data.ag3.Ag3.plot_heterozygosity" title="malariagen_data.ag3.Ag3.plot_heterozygosity"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_heterozygosity</span></code></a>(sample, region, ...[, ...])</p></td>
<td><p>Plot windowed heterozygosity for a single sample over a genome region.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_heterozygosity_track.html#malariagen_data.ag3.Ag3.plot_heterozygosity_track" title="malariagen_data.ag3.Ag3.plot_heterozygosity_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_heterozygosity_track</span></code></a>(sample, region, ...)</p></td>
<td><p>Plot windowed heterozygosity for a single sample over a genome region.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_heatmap.html#malariagen_data.ag3.Ag3.plot_frequencies_heatmap" title="malariagen_data.ag3.Ag3.plot_frequencies_heatmap"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_frequencies_heatmap</span></code></a>(df[, index, ...])</p></td>
<td><p>Plot a heatmap from a pandas DataFrame of frequencies, e.g., output from <cite>Ag3.snp_allele_frequencies()</cite> or <cite>Ag3.gene_cnv_frequencies()</cite>.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_interactive_map.html#malariagen_data.ag3.Ag3.plot_frequencies_interactive_map" title="malariagen_data.ag3.Ag3.plot_frequencies_interactive_map"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_frequencies_interactive_map</span></code></a>(ds[, ...])</p></td>
<td><p>Create an interactive map with markers showing variant frequencies or cohorts grouped by area (space), period (time) and taxon.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_map_markers.html#malariagen_data.ag3.Ag3.plot_frequencies_map_markers" title="malariagen_data.ag3.Ag3.plot_frequencies_map_markers"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_frequencies_map_markers</span></code></a>(m, ds, variant, ...)</p></td>
<td><p>Plot markers on a map showing variant frequencies for cohorts grouped by area (space), period (time) and taxon.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_frequencies_time_series.html#malariagen_data.ag3.Ag3.plot_frequencies_time_series" title="malariagen_data.ag3.Ag3.plot_frequencies_time_series"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_frequencies_time_series</span></code></a>(ds[, height, ...])</p></td>
<td><p>Create a time series plot of variant frequencies using plotly.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_genes.html#malariagen_data.ag3.Ag3.plot_genes" title="malariagen_data.ag3.Ag3.plot_genes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_genes</span></code></a>(region[, width, height, show, ...])</p></td>
<td><p>Plot a genes track, using bokeh.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_snps.html#malariagen_data.ag3.Ag3.plot_snps" title="malariagen_data.ag3.Ag3.plot_snps"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_snps</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Plot SNPs in a given genome region. SNPs are shown as rectangles, with segregating and non-segregating SNPs positioned on different levels, and coloured by site filter. Parameters ---------- region : str     Contig name (e.g., &quot;2L&quot;), gene name (e.g., &quot;AGAP007280&quot;) or     genomic region defined with coordinates (e.g.,     &quot;2L:44989425-44998059&quot;). sample_sets : str or list of str, optional     Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of     sample set identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a     release identifier (e.g., &quot;3.0&quot;) or a list of release identifiers. sample_query : str, optional     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;country == 'Burkina Faso'&quot;. site_mask : str, optional     Site filters mask to apply, e.g. &quot;gamb_colu&quot; width : int, optional     Width of plot in pixels (px). track_height : int, optional     Height of SNPs track in pixels (px). genes_height : int, optional     Height of genes track in pixels (px). max_snps : int, optional     Maximum number of SNPs to show. show : bool, optional     If True, show the plot. Returns ------- fig : Figure     Bokeh figure.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_snps_track.html#malariagen_data.ag3.Ag3.plot_snps_track" title="malariagen_data.ag3.Ag3.plot_snps_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_snps_track</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Plot SNPs in a given genome region.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_transcript.html#malariagen_data.ag3.Ag3.plot_transcript" title="malariagen_data.ag3.Ag3.plot_transcript"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_transcript</span></code></a>(transcript[, width, height, ...])</p></td>
<td><p>Plot a transcript, using bokeh.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.resolve_region.html#malariagen_data.ag3.Ag3.resolve_region" title="malariagen_data.ag3.Ag3.resolve_region"><code class="xref py py-obj docutils literal notranslate"><span class="pre">resolve_region</span></code></a>(region)</p></td>
<td><p>Convert a genome region into a standard data structure.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.site_annotations.html#malariagen_data.ag3.Ag3.site_annotations" title="malariagen_data.ag3.Ag3.site_annotations"><code class="xref py py-obj docutils literal notranslate"><span class="pre">site_annotations</span></code></a>(region[, site_mask, ...])</p></td>
<td><p>Load site annotations.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.site_filters.html#malariagen_data.ag3.Ag3.site_filters" title="malariagen_data.ag3.Ag3.site_filters"><code class="xref py py-obj docutils literal notranslate"><span class="pre">site_filters</span></code></a>(region, mask[, field, ...])</p></td>
<td><p>Access SNP site filters.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_allele_counts.html#malariagen_data.ag3.Ag3.snp_allele_counts" title="malariagen_data.ag3.Ag3.snp_allele_counts"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_allele_counts</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Compute SNP allele counts.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_allele_frequencies.html#malariagen_data.ag3.Ag3.snp_allele_frequencies" title="malariagen_data.ag3.Ag3.snp_allele_frequencies"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_allele_frequencies</span></code></a>(transcript, cohorts)</p></td>
<td><p>Compute per variant allele frequencies for a gene transcript.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_allele_frequencies_advanced.html#malariagen_data.ag3.Ag3.snp_allele_frequencies_advanced" title="malariagen_data.ag3.Ag3.snp_allele_frequencies_advanced"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_allele_frequencies_advanced</span></code></a>(transcript, ...)</p></td>
<td><p>Group samples by taxon, area (space) and period (time), then compute SNP allele counts and frequencies.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_calls.html#malariagen_data.ag3.Ag3.snp_calls" title="malariagen_data.ag3.Ag3.snp_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_calls</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Access SNP sites, site filters and genotype calls.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_dataset.html#malariagen_data.ag3.Ag3.snp_dataset" title="malariagen_data.ag3.Ag3.snp_dataset"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_dataset</span></code></a>(*args, **kwargs)</p></td>
<td><p>Deprecated, this method has been renamed to snp_calls().</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_effects.html#malariagen_data.ag3.Ag3.snp_effects" title="malariagen_data.ag3.Ag3.snp_effects"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_effects</span></code></a>(transcript[, site_mask])</p></td>
<td><p>Compute variant effects for a gene transcript.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_genotypes.html#malariagen_data.ag3.Ag3.snp_genotypes" title="malariagen_data.ag3.Ag3.snp_genotypes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_genotypes</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Access SNP genotypes and associated data.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_variants.html#malariagen_data.ag3.Ag3.snp_variants" title="malariagen_data.ag3.Ag3.snp_variants"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_variants</span></code></a>(region[, site_mask, ...])</p></td>
<td><p>Access SNP sites and site filters.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.snp_sites.html#malariagen_data.ag3.Ag3.snp_sites" title="malariagen_data.ag3.Ag3.snp_sites"><code class="xref py py-obj docutils literal notranslate"><span class="pre">snp_sites</span></code></a>(region, field[, site_mask, ...])</p></td>
<td><p>Access SNP site data (positions and alleles).</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.view_alignments.html#malariagen_data.ag3.Ag3.view_alignments" title="malariagen_data.ag3.Ag3.view_alignments"><code class="xref py py-obj docutils literal notranslate"><span class="pre">view_alignments</span></code></a>(region, sample[, ...])</p></td>
<td><p>Launch IGV and view sequence read alignments and SNP genotypes from the given sample.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="haplotype-data-access">
<h2>Haplotype data access<a class="headerlink" href="#haplotype-data-access" title="Permalink to this heading">#</a></h2>
<table class="autosummary longtable table autosummary">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.h12_calibration.html#malariagen_data.ag3.Ag3.h12_calibration" title="malariagen_data.ag3.Ag3.h12_calibration"><code class="xref py py-obj docutils literal notranslate"><span class="pre">h12_calibration</span></code></a>(contig, analysis[, ...])</p></td>
<td><p>Generate h12 GWSS calibration data for different window sizes.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.h12_gwss.html#malariagen_data.ag3.Ag3.h12_gwss" title="malariagen_data.ag3.Ag3.h12_gwss"><code class="xref py py-obj docutils literal notranslate"><span class="pre">h12_gwss</span></code></a>(contig, analysis, window_size[, ...])</p></td>
<td><p>Run h12 GWSS.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.h1x_gwss.html#malariagen_data.ag3.Ag3.h1x_gwss" title="malariagen_data.ag3.Ag3.h1x_gwss"><code class="xref py py-obj docutils literal notranslate"><span class="pre">h1x_gwss</span></code></a>(contig, analysis, window_size, ...)</p></td>
<td><p>Run a H1X genome-wide scan to detect genome regions with shared selective sweeps between two cohorts. Parameters ---------- contig: str     Chromosome arm (e.g., &quot;2L&quot;) analysis : {&quot;arab&quot;, &quot;gamb_colu&quot;, &quot;gamb_colu_arab&quot;}     Which phasing analysis to use. If analysing only An. arabiensis, the     &quot;arab&quot; analysis is best. If analysing only An. gambiae and An. coluzzii, the &quot;gamb_colu&quot; analysis is best. Otherwise, use the     &quot;gamb_colu_arab&quot; analysis. window_size : int     The size of windows used to calculate h12 over. cohort1_query : str     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. cohort2_query : str     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. sample_sets : str or list of str, optional     Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of     sample set identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a     release identifier (e.g., &quot;3.0&quot;) or a list of release identifiers. cohort_size : int, optional     If provided, randomly down-sample to the given cohort size. random_seed : int, optional     Random seed used for down-sampling. Returns ------- x : numpy.ndarray     An array containing the window centre point genomic positions. h1x : numpy.ndarray     An array with H1X statistic values for each window.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.haplotypes.html#malariagen_data.ag3.Ag3.haplotypes" title="malariagen_data.ag3.Ag3.haplotypes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">haplotypes</span></code></a>(region, analysis[, sample_sets, ...])</p></td>
<td><p>Access haplotype data.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_haplotypes.html#malariagen_data.ag3.Ag3.open_haplotypes" title="malariagen_data.ag3.Ag3.open_haplotypes"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_haplotypes</span></code></a>(sample_set, analysis)</p></td>
<td><p>Open haplotypes zarr.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_haplotype_sites.html#malariagen_data.ag3.Ag3.open_haplotype_sites" title="malariagen_data.ag3.Ag3.open_haplotype_sites"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_haplotype_sites</span></code></a>(analysis)</p></td>
<td><p>Open haplotype sites zarr.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h12_calibration.html#malariagen_data.ag3.Ag3.plot_h12_calibration" title="malariagen_data.ag3.Ag3.plot_h12_calibration"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_h12_calibration</span></code></a>(contig, analysis[, ...])</p></td>
<td><p>Plot h12 GWSS calibration data for different window sizes.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h12_gwss_track.html#malariagen_data.ag3.Ag3.plot_h12_gwss_track" title="malariagen_data.ag3.Ag3.plot_h12_gwss_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_h12_gwss_track</span></code></a>(contig, analysis, ...[, ...])</p></td>
<td><p>Plot h12 GWSS data.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h12_gwss.html#malariagen_data.ag3.Ag3.plot_h12_gwss" title="malariagen_data.ag3.Ag3.plot_h12_gwss"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_h12_gwss</span></code></a>(contig, analysis, window_size)</p></td>
<td><p>Plot h12 GWSS data.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h1x_gwss.html#malariagen_data.ag3.Ag3.plot_h1x_gwss" title="malariagen_data.ag3.Ag3.plot_h1x_gwss"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_h1x_gwss</span></code></a>(contig, analysis, window_size, ...)</p></td>
<td><p>Run and plot a H1X genome-wide scan to detect genome regions with shared selective sweeps between two cohorts. Parameters ---------- contig: str     Chromosome arm (e.g., &quot;2L&quot;) analysis : {&quot;arab&quot;, &quot;gamb_colu&quot;, &quot;gamb_colu_arab&quot;}     Which phasing analysis to use. If analysing only An. arabiensis, the     &quot;arab&quot; analysis is best. If analysing only An. gambiae and An. coluzzii, the &quot;gamb_colu&quot; analysis is best. Otherwise, use the     &quot;gamb_colu_arab&quot; analysis. window_size : int     The size of windows used to calculate h12 over. cohort1_query : str     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. cohort2_query : str     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. sample_sets : str or list of str, optional     Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of     sample set identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a     release identifier (e.g., &quot;3.0&quot;) or a list of release identifiers. cohort_size : int, optional     If provided, randomly down-sample to the given cohort size. random_seed : int, optional     Random seed used for down-sampling. title : str, optional     If provided, title string is used to label plot. width : int, optional     Plot width in pixels (px). track_height : int. optional     GWSS track height in pixels (px). genes_height : int. optional     Gene track height in pixels (px). Returns ------- fig : figure     A plot showing windowed H1X statistic with gene track on x-axis.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_h1x_gwss_track.html#malariagen_data.ag3.Ag3.plot_h1x_gwss_track" title="malariagen_data.ag3.Ag3.plot_h1x_gwss_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_h1x_gwss_track</span></code></a>(contig, analysis, ...[, ...])</p></td>
<td><p>Run and plot a H1X genome-wide scan to detect genome regions with shared selective sweeps between two cohorts. Parameters ---------- contig: str     Chromosome arm (e.g., &quot;2L&quot;) analysis : {&quot;arab&quot;, &quot;gamb_colu&quot;, &quot;gamb_colu_arab&quot;}     Which phasing analysis to use. If analysing only An. arabiensis, the     &quot;arab&quot; analysis is best. If analysing only An. gambiae and An. coluzzii, the &quot;gamb_colu&quot; analysis is best. Otherwise, use the     &quot;gamb_colu_arab&quot; analysis. window_size : int     The size of windows used to calculate h12 over. cohort1_query : str     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. cohort2_query : str     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. sample_sets : str or list of str, optional     Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of     sample set identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a     release identifier (e.g., &quot;3.0&quot;) or a list of release identifiers. cohort_size : int, optional     If provided, randomly down-sample to the given cohort size. random_seed : int, optional     Random seed used for down-sampling. title : str, optional     If provided, title string is used to label plot. width : int, optional     Plot width in pixels (px). height : int. optional     Plot height in pixels (px). show : bool, optional     If True, show the plot. x_range : bokeh.models.Range1d, optional     X axis range (for linking to other tracks). Returns ------- fig : figure     A plot showing windowed H1X statistic across chosen contig.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_haplotype_clustering.html#malariagen_data.ag3.Ag3.plot_haplotype_clustering" title="malariagen_data.ag3.Ag3.plot_haplotype_clustering"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_haplotype_clustering</span></code></a>(region, analysis)</p></td>
<td><p>Hierarchically cluster haplotypes in region and produce an interactive plot. Parameters ---------- region: str or list of str or Region or list of Region     Chromosome arm (e.g., &quot;2L&quot;), gene name (e.g., &quot;AGAP007280&quot;), genomic     region defined with coordinates (e.g., &quot;2L:44989425-44998059&quot;) or a     named tuple with genomic location <cite>Region(contig, start, end)</cite>. Multiple values can be provided as a list, in which case data will     be concatenated, e.g., [&quot;3R&quot;, &quot;3L&quot;]. analysis : {&quot;arab&quot;, &quot;gamb_colu&quot;, &quot;gamb_colu_arab&quot;}     Which phasing analysis to use. If analysing only An. arabiensis, the     &quot;arab&quot; analysis is best. If analysing only An. gambiae and An. coluzzii, the &quot;gamb_colu&quot; analysis is best. Otherwise, use the     &quot;gamb_colu_arab&quot; analysis. sample_sets : str or list of str, optional     Can be a sample set identifier (e.g., &quot;AG1000G-AO&quot;) or a list of     sample set identifiers (e.g., [&quot;AG1000G-BF-A&quot;, &quot;AG1000G-BF-B&quot;]) or a     release identifier (e.g., &quot;3.0&quot;) or a list of release identifiers. sample_query : str, optional     A pandas query string which will be evaluated against the sample     metadata e.g., &quot;taxon == 'coluzzii' and country == 'Burkina Faso'&quot;. color : str, optional     Identifies a column in the sample metadata which determines the colour     of dendrogram leaves (haplotypes). symbol : str, optional     Identifies a column in the sample metadata which determines the shape     of dendrogram leaves (haplotypes). linkage_method: str, optional     The linkage algorithm to use, valid options are 'single', 'complete',     'average', 'weighted', 'centroid', 'median' and 'ward'. See the Linkage     Methods section of the scipy.cluster.hierarchy.linkage docs for full     descriptions. count_sort: bool, optional     For each node n, the order (visually, from left-to-right) n's two descendant     links are plotted is determined by this parameter. If True, the child with     the minimum number of original objects in its cluster is plotted first. Note     distance_sort and count_sort cannot both be True. distance_sort: bool, optional     For each node n, the order (visually, from left-to-right) n's two descendant     links are plotted is determined by this parameter. If True, The child with the     minimum distance between its direct descendants is plotted first. cohort_size : int, optional     If provided, randomly down-sample to the given cohort size. random_seed : int, optional     Random seed used for down-sampling. width : int, optional     The figure width in pixels height: int, optional     The figure height in pixels.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_haplotype_network.html#malariagen_data.ag3.Ag3.plot_haplotype_network" title="malariagen_data.ag3.Ag3.plot_haplotype_network"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_haplotype_network</span></code></a>(region, analysis[, ...])</p></td>
<td><p>Construct a median-joining haplotype network and display it using Cytoscape.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="cnv-data-access">
<h2>CNV data access<a class="headerlink" href="#cnv-data-access" title="Permalink to this heading">#</a></h2>
<table class="autosummary longtable table autosummary">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cnv_coverage_calls.html#malariagen_data.ag3.Ag3.cnv_coverage_calls" title="malariagen_data.ag3.Ag3.cnv_coverage_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">cnv_coverage_calls</span></code></a>(region, sample_set, analysis)</p></td>
<td><p>Access CNV HMM data from genome-wide CNV discovery and filtering.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cnv_discordant_read_calls.html#malariagen_data.ag3.Ag3.cnv_discordant_read_calls" title="malariagen_data.ag3.Ag3.cnv_discordant_read_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">cnv_discordant_read_calls</span></code></a>(contig[, ...])</p></td>
<td><p>Access CNV discordant read calls data.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.cnv_hmm.html#malariagen_data.ag3.Ag3.cnv_hmm" title="malariagen_data.ag3.Ag3.cnv_hmm"><code class="xref py py-obj docutils literal notranslate"><span class="pre">cnv_hmm</span></code></a>(region[, sample_sets, sample_query, ...])</p></td>
<td><p>Access CNV HMM data from CNV calling.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.gene_cnv.html#malariagen_data.ag3.Ag3.gene_cnv" title="malariagen_data.ag3.Ag3.gene_cnv"><code class="xref py py-obj docutils literal notranslate"><span class="pre">gene_cnv</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Compute modal copy number by gene, from HMM data.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.gene_cnv_frequencies.html#malariagen_data.ag3.Ag3.gene_cnv_frequencies" title="malariagen_data.ag3.Ag3.gene_cnv_frequencies"><code class="xref py py-obj docutils literal notranslate"><span class="pre">gene_cnv_frequencies</span></code></a>(region, cohorts[, ...])</p></td>
<td><p>Compute modal copy number by gene, then compute the frequency of amplifications and deletions in one or more cohorts, from HMM data.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.gene_cnv_frequencies_advanced.html#malariagen_data.ag3.Ag3.gene_cnv_frequencies_advanced" title="malariagen_data.ag3.Ag3.gene_cnv_frequencies_advanced"><code class="xref py py-obj docutils literal notranslate"><span class="pre">gene_cnv_frequencies_advanced</span></code></a>(region, ...[, ...])</p></td>
<td><p>Group samples by taxon, area (space) and period (time), then compute gene CNV counts and frequencies.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_cnv_coverage_calls.html#malariagen_data.ag3.Ag3.open_cnv_coverage_calls" title="malariagen_data.ag3.Ag3.open_cnv_coverage_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_cnv_coverage_calls</span></code></a>(sample_set, analysis)</p></td>
<td><p>Open CNV coverage calls zarr.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_cnv_discordant_read_calls.html#malariagen_data.ag3.Ag3.open_cnv_discordant_read_calls" title="malariagen_data.ag3.Ag3.open_cnv_discordant_read_calls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_cnv_discordant_read_calls</span></code></a>(sample_set)</p></td>
<td><p>Open CNV discordant read calls zarr.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.open_cnv_hmm.html#malariagen_data.ag3.Ag3.open_cnv_hmm" title="malariagen_data.ag3.Ag3.open_cnv_hmm"><code class="xref py py-obj docutils literal notranslate"><span class="pre">open_cnv_hmm</span></code></a>(sample_set)</p></td>
<td><p>Open CNV HMM zarr.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage.html#malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage" title="malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_cnv_hmm_coverage</span></code></a>(sample, region[, ...])</p></td>
<td><p>Plot CNV HMM data for a single sample, together with a genes track, using bokeh.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage_track.html#malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage_track" title="malariagen_data.ag3.Ag3.plot_cnv_hmm_coverage_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_cnv_hmm_coverage_track</span></code></a>(sample, region)</p></td>
<td><p>Plot CNV HMM data for a single sample, using bokeh.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap.html#malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap" title="malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_cnv_hmm_heatmap</span></code></a>(region[, sample_sets, ...])</p></td>
<td><p>Plot CNV HMM data for multiple samples as a heatmap, with a genes track, using bokeh.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap_track.html#malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap_track" title="malariagen_data.ag3.Ag3.plot_cnv_hmm_heatmap_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_cnv_hmm_heatmap_track</span></code></a>(region[, ...])</p></td>
<td><p>Plot CNV HMM data for multiple samples as a heatmap, using bokeh.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_roh.html#malariagen_data.ag3.Ag3.plot_roh" title="malariagen_data.ag3.Ag3.plot_roh"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_roh</span></code></a>(sample, region, site_mask, window_size)</p></td>
<td><p>Plot windowed heterozygosity and inferred runs of homozygosity for a single sample over a genome region.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.plot_roh_track.html#malariagen_data.ag3.Ag3.plot_roh_track" title="malariagen_data.ag3.Ag3.plot_roh_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_roh_track</span></code></a>(df_roh, region[, width, ...])</p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="generated/malariagen_data.ag3.Ag3.roh_hmm.html#malariagen_data.ag3.Ag3.roh_hmm" title="malariagen_data.ag3.Ag3.roh_hmm"><code class="xref py py-obj docutils literal notranslate"><span class="pre">roh_hmm</span></code></a>(sample, region, site_mask, window_size)</p></td>
<td><p>Infer runs of homozygosity for a single sample over a genome region.</p></td>
</tr>
</tbody>
</table>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer">
                  <!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="index.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">malariagen_data API</p>
      </div>
    </a>
    <a class="right-next"
       href="generated/malariagen_data.ag3.Ag3.pca.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">malariagen_data.ag3.Ag3.pca</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#basic-data-access">Basic data access</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#sample-metadata-access">Sample metadata access</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#snp-data-access">SNP data access</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#haplotype-data-access">Haplotype data access</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#cnv-data-access">CNV data access</a></li>
</ul>
  </nav></div>

  <div class="sidebar-secondary-item">
  <div class="tocsection sourcelink">
    <a href="_sources/Ag3.rst.txt">
      <i class="fa-solid fa-file-lines"></i> Show Source
    </a>
  </div>
</div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=ac02cc09edc035673794"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=ac02cc09edc035673794"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">
  <p class="copyright">
    
      © Copyright 2022, MalariaGEN.
      <br/>
    
  </p>
</div>
      
        <div class="footer-item">
  <p class="sphinx-version">
    Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 7.1.2.
    <br/>
  </p>
</div>
      
    </div>
  
  
  
    <div class="footer-items__end">
      
        <div class="footer-item"><p class="theme-version">
  Built with the <a href="https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html">PyData Sphinx Theme</a> 0.14.1.
</p></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>